### DISCUSSION TITLE
How to pass message from WPF to web

### DISCUSSION THREAD

**Discussion by phuocminh6464** [cefsharp/CefSharp#4435] [General]
Hi everyone, 

My WPF are loading a website by using Cefsharp and **what I want is** passing message from WPF to web and web is able to receive it immediately. After receiving message, website will clear localStorage. Currently, I know that using function **ExecuteScriptAsync** can do it but I don' t want do that.
`ExecuteScriptAsync("window.localStorage.clear();");`

So is there other ways to pass message to web? I want web always listen WPF and can receive message immediately after WPF sending. Something works in JS side same as **JavascriptMessageReceived** event in C# side

**Comment by amaitland** [comment]
> I know that using function **ExecuteScriptAsync** can do it but I don' t want do that.

What is the problem with executing javascript exactly?

> So is there other ways to pass message to web?

You can use a `IJavascriptCallback` if you like. See https://github.com/cefsharp/CefSharp/blob/cefsharp/111/CefSharp.Test/Javascript/JavascriptCallbackTests.cs for examples

> I want web always listen WPF and can receive message immediately after WPF sending.

How is this any different to calling a function directly? Event handlers are really just functions themselves.

> Something works in JS side same as **JavascriptMessageReceived** event in C# side

You could easily implement this in your own app.

**Reply by phuocminh6464** [reply]
> You can use a `IJavascriptCallback` if you like. See https://github.com/cefsharp/CefSharp/blob/cefsharp/111/CefSharp.Test/Javascript/JavascriptCallbackTests.cs for examples

Do following this and it works, thanks you.
